### PR TITLE
Add E2E tests for the Filter Products By Price Block

### LINKED ISSUE #6323: Filter Products by Stock Block: Implement E2E tests
Add E2E tests for the Filter Products by Attribute Block. We should check that:
- when the filter is applied only the products that match that filter are rendered (on post/page and a PHP classic template).
- only on the classic PHP template the refresh is triggered.
- after the page refreshes, the URL is what we expect.
- when the filter button is enabled the page refresh is only triggered once the user clicks on it.

### LINKED ISSUE #6363: Filter Products by Price Block: Implement E2E tests
Add E2E tests for the Filter Products by Price Block. We should check that:

- when the filter is applied only the products that match that filter are rendered (on post/page and a PHP classic template).
- only on the classic PHP template the refresh is triggered.
- after the page refreshes, the URL is what we expect.
- when the filter button is enabled the page refresh is only triggered once the user clicks on it.

### PR DESCRIPTION
**Pull request by gigitux**
This PR adds E2E tests for the Filter Products By Price Block. These tests check that:
- when the filter is applied only the products that match that filter are rendered.
- only on the classic PHP template the refresh is triggered.
- after the page refreshes, the URL is what we expect.
- when the filter button is enabled the page refresh is only triggered once the user clicks on it.

Fixes #6363 

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

#### User Facing Testing

Check out this branch

1. Run `npm run test:e2e tests/e2e/specs/shopper/filter-products-by-price.test.ts`.
2. Be sure that tests don't fail.

* [x] Do not include in the Testing Notes

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+	page.waitForSelector( selectors.frontend.productsList + '.is-loading', {
+		hidden: true,
+	} );
+
+const goToShopPage = () =>
+	page.goto( BASE_URL + '/shop', {
+		waitUntil: 'networkidle0',
+	} );
+
+const { selectors } = block;
```
Comment (tomasztunik) [inline tests/e2e/specs/shopper/filter-products-by-price.test.ts:46]: same issue with the order

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+			const products = await page.$$( selectors.frontend.productsList );
+
+			expect( products ).toHaveLength( 5 );
+		} );
+
+		it( 'should show only products that match the filter', async () => {
+			const isRefreshed = jest.fn( () => void 0 );
+			page.on( 'load', isRefreshed );
+			await setMaxPrice();
+			await page.waitForTimeout( 1000 );
```
Comment (tomasztunik) [inline tests/e2e/specs/shopper/filter-products-by-price.test.ts]: is there a way to replace that with something more meaningful and reliable than time? 

What are we waiting here for? for the `setMaxPrice` to work correctly? can we add something inside that helper to wait for ie. updated element state, idle network or something that makes sense?

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+			const products = await page.$$(
+				selectors.frontend.classicProductsList
+			);
+
+			const pageURL = page.url();
+			const parsedURL = new URL( pageURL );
+
+			expect( isRefreshed ).toBeCalledTimes( 1 );
+			expect( products ).toHaveLength( 1 );
```
Comment (tomasztunik) [inline tests/e2e/specs/shopper/filter-products-by-price.test.ts:156]: should we check here if the expected product is listed?

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+		it( 'should show only products that match the filter', async () => {
+			const isRefreshed = jest.fn( () => void 0 );
+			page.on( 'load', isRefreshed );
+			await setMaxPrice();
+			await page.waitForTimeout( 1000 );
+			await waitForAllProductsBlockLoaded();
+			const products = await page.$$( selectors.frontend.productsList );
+
+			expect( isRefreshed ).not.toBeCalled();
+			expect( products ).toHaveLength( 1 );
```
Comment (tomasztunik) [inline tests/e2e/specs/shopper/filter-products-by-price.test.ts:95]: let's check if expected product is there

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts [hunk 5]
```diff
@@ -87,7 +87,7 @@ describe( `${ block.name } Block`, () => {
 			const isRefreshed = jest.fn( () => void 0 );
 			page.on( 'load', isRefreshed );
 			await setMaxPrice();
-			await page.waitForTimeout( 1000 );
+			await page.waitForNetworkIdle();
```
Comment (tomasztunik) [inline tests/e2e/specs/shopper/filter-products-by-price.test.ts:90]: should we move it inside of setMaxPrice() - this way `setMaxPrice` will ensure the filter was applied.

Comment (gigitux) [reply tests/e2e/specs/shopper/filter-products-by-price.test.ts:90]: I'm not sure. For the PHP template, the setMaxPrice function triggers a page refresh

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts (hunk 3 above)
Comment (gigitux) [reply tests/e2e/specs/shopper/filter-products-by-price.test.ts:156]: [We check it 4 lines below](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6373/files/db0f316f85d061bc353142befce62f3378ea64c6#diff-008c918aa7487e2ab53be1c7032a692c9a904a019933eaef84634a0ff19f12fcR160)

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts (hunk 4 above)
Comment (gigitux) [reply tests/e2e/specs/shopper/filter-products-by-price.test.ts:95]: We check it on the line below

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts (hunk 5 above)
Comment (tomasztunik) [reply tests/e2e/specs/shopper/filter-products-by-price.test.ts:90]: ah, alright :)

File: tests/e2e/specs/shopper/filter-products-by-price.test.ts (hunk 1 above)
Comment (tomasztunik) [inline tests/e2e/specs/shopper/filter-products-by-price.test.ts]: selectors 🆙: 🆙  🆙

**Review by tomasztunik — APPROVED** [review]

**Comment by gigitux** [comment]
I'm merging this PR because the failed tests are unrelated to the changes of this PR.

**Pull request merged**